### ISSUE TITLE
nestjs can't resolve RedisService dependency if not in root module

### ISSUE THREAD

**Issue by daramasala** [kyknow/nestjs-redis#8]
Hi,
I'm registering `RedisModule` in my app module and in another module I have a provider that has the `RedisService` as dependency.

When I start the server, I get an error message:
```
Error: Nest can't resolve dependencies of the MyService (?). Please make sure that the argument at index [0] is available in the OtherModule context.
```

Looks like nestjs can't find the `RedisService` provider. 

Here is the code:
`app.module.ts`
```JS
import { Module } from '@nestjs/common';
import { AppController } from './app.controller';
import { AppService } from './app.service';
import { OtherModule } from './other/other.module';
import { RedisModule } from 'nestjs-redis';

@Module({
  imports: [RedisModule.register({ host: 'localhost' }), OtherModule],
  controllers: [AppController],
  providers: [AppService],
})
export class AppModule {}
```

And in `other.module.ts`:
```JS
import { Module } from '@nestjs/common';
import { MyService } from './my/my.service';

@Module({
  providers: [MyService]
})
export class OtherModule {}

```

And in `my-service.ts`:
```JS
import { Injectable } from '@nestjs/common';
import { RedisService } from 'nestjs-redis';

@Injectable()
export class MyService {
  constructor(redisService: RedisService) {}
}
```

**Comment by skunight** [comment]
Please import this module in 'OtherModule'

**Comment by daramasala** [comment]
1. I tried importing `RedisModule` in `OtherModule`. Doesn't work.
2. I also tried adding the `RedisService` to `OtherModule` providers. Doesn't work.
3. The `RedisCoreModule` is annotated with `@Global()` so its providers should have been available everywhere without importing, no?

Here is a link to a repo which reproduces the problem: https://github.com/daramasala/reproduce-nestjs-redis-not-found
